### PR TITLE
Warn user if ~/.ocean is owned by root

### PR DESCRIPTION
**Pull request by tmarble**
Reminder that #100 is not yet fixed

## Description

Adds warning to start_ocean.sh if ~/.ocean is owned by root

## Is this PR related with an open issue?

Related to Issue #100

### CODE REVIEW

File: start_ocean.sh [hunk 1]
```diff
@@ -63,6 +66,16 @@ function show_banner {
     echo ""
 }
 
+function check_if_issue_100_fixed {
```
Comment (aaitor) [inline start_ocean.sh]: Maybe a function name like `check_if_owned_by_root` would be more clear

**Comment by ttmc** [comment]
@tmarble How are you connected with Ocean Protocol? Are you an employee of (or contractor for) BigchainDB GmbH or DEX? Is this pull request to fulfill a bounty?

**Comment by tmarble** [comment]
@ttmc I work for Informatique, Inc. -- we are collaborating with DEX to ensure the initial developer experience with the Ocean Network is fantastic.

Is there a bounty program?

**Comment by ttmc** [comment]
The reason I asked is that we haven't (yet) established a process for integrating code (copyrighted intellectual property) when the contribution isn't from an employee or someone who agreed to the Terms & Conditions of the services that manage [the Ocean Protocol bounty programs](https://docs.oceanprotocol.com/concepts/bounties/).

If you're doing official contract work for DEX, then the terms of that contract should be sufficient to ensure that your contributions are licensed to DEX (which in turn can license them Apache v2). If not, then we'll have to figure out what to do. _Please let us know._

We value your contribution, but we also need to make sure that the project isn't jeopardized by snippets of unlicensed code.

P.S. It's the end of the day here in Berlin, and Friday is a public holiday here, so it might be a day or two before I can spend some time to propose a resolution (if that is necessary).

**Comment by tmarble** [comment]
@ttmc as someone who is very much involved in the open source legal community
(FFI https://www.fosdem.org/2019/schedule/track/legal_and_policy_issues/ ) I am hopeful that
you will use a simple inbound=outbound licensing strategy (as opposed to a Contribution License Agreement or Copyright Assignment Agreement).

You can find an example of such an approach in this [CONTRIBUTING](https://github.com/dollabs/pamela/blob/master/CONTRIBUTING.md) file. This is not legal advice.

It's an exciting time for Ocean Protocol!

**Comment by ttmc** [comment]
@tmarble It's an incredible stroke of luck that you are our first external contributor (AFAIK), since you know a lot about handling the licensing of code from external contributors.

I looked at those links you shared and was delighted because I had already been thinking along similar lines. Last year I came across the Developer Certificate of Origin via the Hyperledger (blockchain) projects and the more I read about it, the more interested I became in using it for Ocean Protocol.

In December, I drafted some process documents for Ocean Protocol external contributors, and they've been sitting around as pull requests since then. See:

- https://github.com/oceanprotocol/docs/pull/118
- https://github.com/oceanprotocol/dev-ocean/pull/117

Today I updated those pull requests to make the process even simpler. They outline what you would have to do (as an external contributor) before we can merge this pull request... _if the Ocean Protocol team is okay with the proposed process_. I will have to ask them next week.

If you have time, I would appreciate any comments or feedback you might have on those two pull requests (i.e. on the process they describe).

**Comment by tmarble** [comment]
@ttmc Thank you for kind words! Here is my review:

## oceanprotocol/docs#118

It is noted that This is part of fulfilling [oceanprotocol/dev-ocean#104](https://github.com/oceanprotocol/dev-ocean/issues/104)

In that issue there is a reference to [BigchainDB Contributor License Agreement (CLA)](https://www.bigchaindb.com/cla/) 

My opinion is that having a CLA (or CAA) seriously hampers the 
community success of an open source project. This has been 
debated vigorously over the past several years. Perhaps more apropos
for the question at hand is that the Ocean Protocol is intended
to be a high level API an abstracted from specific implementation
details (such as the implementation of smart contracts). Thus for
this reason alone I can recommend against tying Ocean Protocol
contributions to a specific implementation. And, indeed, I recommend
against CLA's generally.

I'm not super sure what "until the external contributor has added the license
[and included a Signed-off-by line] in their commit messages." means.
Does this mean _each_ commit must include the full text of the license?
Does it mean that each commit must include an assertion and/or 
reference to the license by name? I suspect that such interpretation
is overkill. My understanding of the DCO (indeed the value of it) is
that (as should be documented in the project contribution guidelines) it makes
explicit that the contribution is done under the same license as
the project and the contributor has the rights to make the contribution.

Adding the signed off line should be fairly easy with many tools
that know about VCS's (git, in particular). For example magit for Emacs
has support:
https://magit.vc/manual/magit/Editing-Commit-Messages.html#index-git_002dcommit_002dsignoff

There is a comment that [OEP-1 (C4)](https://github.com/oceanprotocol/OEPs/tree/master/1) will have to be updated to reflect this policy.
I would like to highlight that:
- §2.1.6 Seems unnecessary to state "Maintainers are not developers"
- §2.2.3 Needs to be removed
- §2.3.1 Is there a rationale for a "real names policy"? The github user id is a _de facto_ alias
- §2.3.4 What is the process for incorporating third party code?
- §2.4.5 Asking for problems to provable (i.e. reproduced) is a laudable goal, but may discourage users from recording otherwise useful bug information
- §2.4.20 If long standing issues are deleted (instead of being archived) then valuable knowledge may be lost

Other issues to consider
- What does it take to become a Maintainer?
- Should there be a code of conduct?

Oops.. I see the disposition C4 has changed... 
  https://github.com/oceanprotocol/OEPs/pull/148/files
not sure if my comments above are still apropos (or if there is 
a new proces to replace C4)? To whatever degree you're current thinking
includes sorting developers into "internal" and "external" categories I would
encourage the process to have the internal developers follow the
same guidelines insomuch as possible (i.e. eat your own dog food!).

## oceanprotocol/dev-ocean#117

Here are some specific comments on [diff](https://github.com/oceanprotocol/dev-ocean/pull/117/files#diff-056ca6f4dc873f1c5fe708fee73bc491R32)

- Line 32: In general we want license clarity, but to avoid doubt about Ocean related
  projects you probably want to say "under the Apache License version 2.0"
- Line 37: You probaby want to add the "circle C" and year to the copyright assertion, as in "Copyright (c) 2019"
- Line 37: Using a "Gilligan's Island" style phrase to identify the copyright holder(s) probably wants to have the list of holders clearly documented somewhere (e.g. in a CREDITS or CONTRIBUTORS file or something at the top of the repository)

Disclaimers:
**IANAL**: I am not a lawyer
**TINLA**: This is not legal advice

**Comment by ttmc** [comment]
@tmarble 

- The proposed policy for external contributors does not use a CLA. It uses the DCO and Signed-off-by lines in Git commit messages.
- Until now, we (Ocean Protocol developers) haven't been including any copyright notice or SPDX license identifier comment in our source code files. Unfortunately, the DCO refers to "the open source license indicated in the file" so that must be added if it is not already present. (The DCO says: "Everyone is permitted to copy and distribute verbatim copies of this license document, but changing it is not allowed.")
- I think the instructions are quite clear. The only things an external developer must "do" besides read and understand stuff, is 1) add the copyright notice and SPDX line (comments) to the files they added or modified and 2) add a Signed-off-by line to all their Git commit messages. (I also gave some help with how to change old commit messages.)
- I have a pull request to delete OEP-1 (C4) because we never used it anyway. See https://github.com/oceanprotocol/OEPs/pull/148
- Ocean Protocol has a Code of Conduct at https://docs.oceanprotocol.com/concepts/code-of-conduct/ but maybe we should mention it in more places. I made a note to do that.
- I will change that opening paragraph to be more specific about the open source license, i.e. to change "we need you to sign off on licensing your code under an open source license." to "we need you to sign off on licensing your code under an Apache License version 2.0."
- I don't understand the need for the circle (C) and the copyright year. That seems like an artifact from the past. The Git history has a complete, timestamped record of who contributed what and when.
- The same is true of the list of contributors: the entire thing is available, in detail, in the Git (and GitHub) history.

**Comment by ttmc** [comment]
As noted in [the docs pull request](https://github.com/oceanprotocol/docs/pull/118), in the future, everyone will have to include the copyright notices and the Signed-off-by lines but that's not necessary for now and we're in a bit of a crunch period right now, so it's not a good time to be introducing new policies for all developers.

**Comment by tmarble** [comment]
@ttmc replying to your bullets (in order):

1. Great!
2. You are correct that (to the degree feasible) each file in the repo should have a copyright notice and license terms. I may have misunderstood the statement I read about adding the license to _commit messages_ to be taken literally when you probably meant that each file in the commit should have a proper copyright notice.
3. Ok.
4. Ok.
5. Great!
6. Ok.
7. You may want to ask your counsel for advice here.. This has more to do with legal convention around copyright notices than the mechanics of git history (which, obviously, records each contribution). See also https://www.legalzoom.com/articles/how-do-i-use-the-copyright-symbol , https://blogs.findlaw.com/law_and_life/2014/02/what-do-copyright-trademark-symbols-mean.html
8. Ok.

Indeed I am well aware of the crunch period!!!

**Comment by ttmc** [comment]
@tmarble I made a note to ask our counsel about what to put in the copyright line, but that probably won't have a quick turnaround. Meanwhile, let's proceed with the new process...

<hr>

Before we can merge this pull request, we need you to sign off on licensing your code under the [Apache License version 2.0](https://www.apache.org/licenses/LICENSE-2.0). One of the big concerns for people using and developing open source software is that someone who contributed to the code might claim the code infringes on their copyright or patent. To guard against this, we ask all our contributors to take certain steps (detailed below). This gives us the right to use the code contributed and any patents the contribution relies on. It also gives us and our users comfort that they won't be sued for using open source software. We know it's a hassle, but it makes the project more reliable in the long run. Thank you for your understanding and your contribution!

1. Make sure that every file you modified or created contains a copyright notice comment like the following (at the top of the file):

   ```text
   # Copyright Ocean Protocol contributors
   # SPDX-License-Identifier: Apache-2.0
   ```

   If a copyright notice is not present, then add one. Make sure you're using the correct syntax for comments (which varies from language to language). The example shown above is for a Python file.
1. Read the [Developer Certificate of Origin, Version 1.1](https://developercertificate.org/).
1. You will be asked to include a Signed-off-by line in all your commit messages. (Instructions are given in the next step.) Make sure you understand that including a Signed-off-by line in your commits certifies that you can make the statements in the Developer Certificate of Origin. If you have questions about this, then please ask them in the comments below. Do not continue until you fully understand.
1. Make sure that every commit message in this pull request includes a Signed-off-by line of the form:

   ```text
   Signed-off-by: Random J Developer <random@developer.example.org>
   ```

   with your real name and your real email address. Sorry, no pseudonyms or anonymous contributions. Tip: You can tell Git to include a Signed-off-by line in a commit message by using `git commit --signoff` or `git commit -s`. If you didn't include a Signed-off-by line in your commits so far, you have several options. Probably the simplest is to close this pull request on GitHub, then on your local machine [do an interactive rebase of the last n commits](https://help.github.com/en/articles/changing-a-commit-message) (e.g. the last 3 commits, if your pull request contains 3 commits), then rename your local Git branch (`git branch -m <new-name>`), then push the newly-named branch to GitHub to create a new pull request.

**Comment by tmarble** [comment]
@ttmc Ok.. let me know if you want me to revise anything else. Sorry I forgot to turn on my GPG verification too (I'll get that next time)!

**Pull request merged**